misk/src/main/kotlin/misk/web/marshal/Marshaller.kt

```diff
@@ -14,7 +15,15 @@ interface Marshaller<in T> {
   fun contentType(): MediaType?
 
   /** @return The object marshalled into a [ResponseBody] */
-  fun responseBody(o: T): ResponseBody
+  fun responseBody(o: T): ResponseBody = responseBody(o, Headers.Builder())
+
+  /**
+   * Alternate way to marshal the response body with access to response headers.
+   * Invokes [responseBody] by default.
+   *
+   * @return The object marshalled into a [ResponseBody]
+   */
+  fun responseBody(o: T, responseHeaders: Headers.Builder): ResponseBody
```

Comment:
Does this need to return the resulting mutated headers along with the ResponseBody ie. in a `Pair<Headers, ResponseBody>`?

Reply:
My thinking was that the implementation would modify the builder as desired and the caller would update `httpCall.responseHeaders = responseHeaders.build()` after invoking this method. The strategies seem pretty equivalent and I think it's a bit cleaner as-is.

Of course, this all depends on `httpCall.responseHeaders` being mutable, which doesn't seem to be the case, so I'm not sure if this approach will work at all.